src/features/callAssignments/models/CallAssignmentModel.spec.ts

```diff
@@ -0,0 +1,244 @@
+              id: 2,
+              target: {
+                filter_spec: [
+                  {
+                    config: {},
+                    type: FILTER_TYPE.ALL,
+                  },
+                ],
+                id: 101,
+              },
+              title: 'My assignment',
+            },
+          ]),
+          statsById: {},
+        },
+      });
+
+      when(mockClient.get<CallAssignmentStats>(anything())).thenResolve({
+        allTargets: 100,
+        allocated: 0,
+        blocked: 0,
+        callBackLater: 0,
+        calledTooRecently: 0,
+        done: 50,
+        id: 2,
+        missingPhoneNumber: 0,
+        organizerActionNeeded: 0,
+        queue: 0,
+        ready: 50,
+      });
+
+      const apiClient = instance(mockClient);
+
+      const env = new Environment(store, apiClient);
+      const model = new CallAssignmentModel(env, 1, 2);
+
+      const future = model.getStats();
+
+      expect(future.data).toEqual({
+        allTargets: 0,
+        allocated: 0,
+        blocked: 0,
+        callBackLater: 0,
+        calledTooRecently: 0,
+        done: 0,
+        id: 2,
+        missingPhoneNumber: 0,
+        organizerActionNeeded: 0,
+        queue: 0,
+        ready: 0,
+      });
+      expect(future.isLoading).toBeTruthy();
+      expect(future.error).toBeNull();
+    });
+
+    it('returns assignment stats when loaded', () => {
+      const store = createStore({
+        callAssignments: {
+          assignmentList: mockList<CallAssignmentData>([
+            {
+              cooldown: 3,
+              id: 2,
+              target: {
+                filter_spec: [
+                  {
+                    config: {},
+                    type: FILTER_TYPE.ALL,
+                  },
+                ],
+                id: 101,
+              },
+              title: 'My assignment',
+            },
+          ]),
+          statsById: {
+            2: mockItem({
+              allTargets: 100,
+              allocated: 0,
+              blocked: 0,
+              callBackLater: 0,
+              calledTooRecently: 0,
+              done: 50,
+              id: 2,
+              missingPhoneNumber: 0,
+              organizerActionNeeded: 0,
+              queue: 0,
+              ready: 50,
+            }),
+          },
+        },
+      });
+
+      const apiClient = instance(mockClient);
+
+      const env = new Environment(store, apiClient);
+      const model = new CallAssignmentModel(env, 1, 2);
+
+      const future = model.getStats();
+
+      expect(future.data).toEqual({
+        allTargets: 100,
+        allocated: 0,
+        blocked: 0,
+        callBackLater: 0,
+        calledTooRecently: 0,
+        done: 50,
+        id: 2,
+        missingPhoneNumber: 0,
+        organizerActionNeeded: 0,
+        queue: 0,
+        ready: 50,
+      });
+      expect(future.isLoading).toBeFalsy();
+      expect(future.error).toBeNull();
+    });
+  });
+});
```

Comment:
The structure in this test does go against the structure used in the Great Test Refactor, where we wanted to build test files that have only one "describe" clause, followed by "it" clauses. Does the testing of this new logic require more complex test structure? Or is it a habit, and can be re-structured?

Reply:
This is a very valid point! I'm not sure I mind having multiple `describe()` when they very clearly relate to separate parts of the program code, like in this case where the other `describe()` is for the class, and then we have one for each tested piece of interface (property/method).

Let's chat about how it could be refactored after merging, if we want to. 💯